### ISSUE TITLE
Move from docker Rigetti toolchain to local executables

### ISSUE THREAD

**Issue by crazy4pi314** [unitaryfund/mitiq#897]
In a recent upgrade, pyquil and the Forest SDK now have native executables built for all operating systems that act as the local simulation and build toolchain. Right now in both our CI and local dev experiences we need Docker locally installed and running with ports configured correctly to make any testing or use of pyquil work. We should make a plan to migrate to the native executable install path for both our CI and the developer instructions for Mitiq.

**Comment by natestemen** [comment]
Looking through the pyquil [installation docs](https://pyquil-docs.rigetti.com/en/stable/start.html), it seems like the downloads are all manual installations. Of course, one can `curl`/`wget` the resource, then use the instructions from the installations docs in our [`build.yml`](https://github.com/unitaryfund/mitiq/blob/65879829d41944ed11a1fef4babe2c0a0e8e236c/.github/workflows/build.yml) workflow, but that seems like it might be less stable than using the docker image. Is there another download that I might be missing somewhere?

cc @willzeng in case you're still plugged in here, and know of an easier download mechanism.

**Comment by cosenal** [comment]
As a byproduct, this will fix related warnings that are triggered on a `make docs` execution. 
```
/Users/cosenal/Coding/unitaryfund/mitiq/docs/source/examples/vqe-pyquil-demo.ipynb: WARNING: Executing notebook failed: CellExecutionError [mystnb.exec]
/Users/cosenal/Coding/unitaryfund/mitiq/docs/source/examples/vqe-pyquil-demo.ipynb: WARNING: Notebook exception traceback saved in: /Users/cosenal/Coding/unitaryfund/mitiq/docs/build/html/reports/examples/vqe-pyquil-demo.err.log [mystnb.exec]
/Users/cosenal/Coding/unitaryfund/mitiq/docs/source/examples/pyquil_demo.ipynb: WARNING: Executing notebook failed: CellExecutionError [mystnb.exec]
/Users/cosenal/Coding/unitaryfund/mitiq/docs/source/examples/pyquil_demo.ipynb: WARNING: Notebook exception traceback saved in: /Users/cosenal/Coding/unitaryfund/mitiq/docs/build/html/reports/examples/pyquil_demo.err.log [mystnb.exec]
```

**Comment by cosenal** [comment]
I agree with @natestemen here. I don't see much value in migrating to local executables.
The step of pulling the rigetti Docker images in CI lasts ~40s, which is negligible compared to the rest of the workflow. It pulls the latest versions of qvm/quilc and it's just two lines of code, as opposed to many more if we go with installing the debian package, see for example https://github.com/entropicalabs/openqaoa/blob/2cc9ea00c0d0c460cfa287e6e0d0ef05dbfb764b/.github/workflows/test_dev_ext.yml#L27-L42

Closing this, but you're welcome to reopen it if there is an explicit value of replacing Docker with the executables.
